DirectProgramming/DPC++/StructuredGrids/1d_HeatTransfer/src/1d_HeatTransfer.cpp

```diff
@@ -60,114 +60,177 @@ void Usage(string programName) {
-              } else {
-                temperature_next[k] =
-                    C * (temperature[k + 1] - 2 * temperature[k] + temperature[k - 1]) +
-                    temperature[k];
-              }
-            });  // end parallel for loop in kernel1
-          });    // end device queue
-
-        } else {
-          q.submit([&](handler& h) {
-            // The size of memory amount that will be given to the buffer.
-            range<1> num_items{num_p + 2};
-
-            accessor temperature(temperature_buf, h);
-            accessor temperature_next(temperature_next_buf, h);
-
-            h.parallel_for(num_items, [=](id<1> k) {
-              size_t gid = k.get(0);
-
-              if (gid == 0) {
-              } else if (gid == num_p + 1) {
-                temperature[k] = temperature_next[k - 1];
-              } else {
-                temperature[k] = C * (temperature_next[k + 1] - 2 * temperature_next[k] +
-                                  temperature_next[k - 1]) +
-                             temperature_next[k];
-              }
-            });  // end parallel for loop in kernel2
-          });    // end device queue
-        }        // end if %2
-      }          // end iteration
-    }            // end buffer scope
-
-    q.wait_and_throw();
+  ofstream err_file;
+  err_file.open(path);
 
-  } catch (sycl::exception e) {
-    cout << "SYCL exception caught: " << e.what() << "\n";
+  err_file << " \t idx\theat[i]\t\theat_CPU[i] \n";
+
+  for (unsigned int i = 0; i < num_point + 2; i++) {
+    err_file << "\n RESULT: " << i << "\t" << std::setw(12) << std::left
+             << device_results[i] << "\t" << host_results[i];
+
+    difference = fabsf(host_results[i] - device_results[i]);
+    norm2 += difference * difference;
+
+    if (difference > delta) {
+      err = true;
+      err_file << ", diff: " << difference;
+    }
   }
 
-  if (i % 2 != 0)
-    return arr;
+  if (err == true)
+    cout << "  FAIL! Please check " << path << "\n";
   else
-    return arr_next;
+    cout << "  PASSED!\n";
+  return err;
+}
+
+//
+// Compute heat on the device
+//
+void ComputeHeatDevice(float C, unsigned int num_p, unsigned int num_iter,
+		       float* arr_CPU) {
+  // Define the device queue
+  queue q;
+  cout << "Device\n";
+  cout << "  Kernel runs on " << q.get_device().get_info<info::device::name>() << "\n";
+
+  // Temperatures of the current and next iteration
+  float* arr_host = new float[num_p + 2];
+  float* arr_next_host = new float[num_p + 2];
+
+  Initialize(arr_host, arr_next_host, num_p + 2);
+
+  buffer arr_buf(arr_host, range(num_p + 2));
+  buffer arr_next_buf(arr_next_host, range(num_p + 2));
+
+  // Start timer
+  dpc_common::TimeInterval t_par;
+
+  // Iterate over timesteps
+  for (int i = 1; i <= num_iter; i++) {
+    q.submit([&](auto& h) {
+	       accessor arr(arr_buf, h);
+	       accessor arr_next(arr_next_buf, h);
+
+	       auto step =
+		 [=](id<1> k) {
+		   
```

Comment:
Extra blank line.

Reply:
Both lines?

Reply:
I was talking about line 138.   The blank line at 135 is more normal to separate declarations from code.